modules/server/src/main/ts/bedrock/server/Serve.ts

```diff
@@ -86,31 +85,37 @@ export const startCustom = async (settings: ServeSettings, createServer: (listen
       stopPort: 20000
     });
 
-    const server = createServer((request, response) => {
+    const server = createServer(port, (request, response) => {
       const done = finalhandler(request, response);
       Routes.route(routers, fallback, request, response, done);
-    }).listen(port);
+    });
+    await server.start();
 
     return {
       port,
-      server,
       markLoaded: api.markLoaded,
       enableHud: api.enableHud,
       awaitDone: api.awaitDone,
-      shutdown: () => {
-        return new Promise<void>((resolve) => {
-          server.close();
-          // TODO: Find out why this doesn't shutdown quickly as we may not be closing connections properly
-          // For now though give the server 1 sec to shutdown gracefully
-          setTimeout(resolve, 1000);
-        });
-      }
+      shutdown: server.stop
     };
   } catch (err) {
     return Promise.reject('Error looking for open port between 8000 and 20000: ' + err);
   }
 };
 
 export const start = (settings: ServeSettings): Promise<ServeService> => {
-  return startCustom(settings, http.createServer);
+  return startCustom(settings, (port, listener) => {
+    const server = http.createServer(listener);
+    return {
+      start: () => {
+        server.listen(port);
+        return Promise.resolve();
+      },
```

Comment:
Not sure how important this is since setting up a listening socket is probably very fast. But it feels more correct to use the async callback for this one that just resolving it.

```suggestion
      start: () => {
        return new Promise((resolve) => {
          server.listen(port, resolve);
        });
      },
```

Reply:
Good catch, thanks Spocke!

Reply:
Worth noting we weren't waiting for this previously either 😢